feflow/protocols/nonequilibrium_cycling.py

```diff
@@ -1039,3 +1010,26 @@ def _gather(
 
         # This can be populated however we want
         return outputs
+
+    # TODO: Maybe this could be a utility function. Is this something protocol-specific?
```

Comment:
It's mostly specific to this Protocol - also this should go in the Protocol validate method.